test/jdk/java/awt/ScrollPane/ScrollPaneWindowsTest.java

```diff
@@ -0,0 +1,184 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4452612
+  @requires os.family=="windows"
+  @summary The popup menu of the scroll bar doesn't work properly in Window2000.
+  @key headful
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Insets;
+import java.awt.Panel;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+import java.awt.ScrollPaneAdjustable;
+
+import java.awt.event.AdjustmentListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+public class ScrollPaneWindowsTest implements AdjustmentListener {
+    ScrollPane sp;
+    Panel p;
+    Robot robot;
+    Frame frame;
+    Insets paneInsets;
+    public static final Object LOCK = new Object();
+    ScrollPaneAdjustable vScroll;
+    ScrollPaneAdjustable hScroll;
+    boolean notifyReceived = false;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneWindowsTest scrollTest = new ScrollPaneWindowsTest();
+        scrollTest.init();
+        scrollTest.start();
+    }
+
+    public void init() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame("ScrollPaneWindowsTest");
+            frame.setLayout(new BorderLayout(1, 1));
+            p = new Panel();
+            p.setLayout(null);
+            p.setSize(new Dimension(800, 800));
+            sp = new ScrollPane(ScrollPane.SCROLLBARS_ALWAYS);
+            vScroll = (ScrollPaneAdjustable) sp.getVAdjustable();
+            hScroll = (ScrollPaneAdjustable) sp.getHAdjustable();
+        });
+        robot = new Robot();
+
+        vScroll.addAdjustmentListener(this);
+        hScroll.addAdjustmentListener(this);
+    }
+
+    public void start() throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                sp.add(p);
+                frame.add(sp);
+                frame.pack();
+                frame.setSize(400, 400);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+                paneInsets = sp.getInsets();
+                System.out.println("Insets: right = " + paneInsets.right + " bottom =  " + paneInsets.bottom);
+            });
+
+            robot.delay(100);
+            robot.waitForIdle();
+
+            robot.mouseMove(sp.getLocationOnScreen().x + sp.getWidth() - paneInsets.right / 2,
+                    sp.getLocationOnScreen().y + sp.getHeight() / 2);
```

Comment:
You should get the location on the scroll pane and its size inside `invokeAndWait` and use these two new fields to move the mouse.